### ISSUE TITLE
[Accessibility]: Inconsistent hint text location in VADS components

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/vets-design-system-documentation#5489]
### Point of contact

Jamie Klenetsky Fay

### How severe do you think this issue is?

High

### Severity rationale

Screen reader users in forms mode might miss critical instructions in hint text when certain VADS components are used.

### URL where issue occurs

https://design.va.gov/storybook/?path=/docs/about-introduction--docs (across the entire design system)

### Component or pattern (if applicable)

Some VADS components put the hint text inside the label or legend:
- Checkbox group
- Radio button
- Telephone input
- Text input
- Textarea

Some put them beneath the label or legend:
- Checkbox
- Date input
- File input multiple
- File input
- Memorable date
- Select

### Environment details

All

### Steps to reproduce

Inspect the hint text of all VADS components that offer a hint text prop.

### Observed behavior

To demonstrate the issue with hint text that's not included in the label/legend, go to https://design.va.gov/storybook/?path=/story/uswds-va-checkbox--with-hint-text, turn on VoiceOver, and tab to the checkbox input. VoiceOver doesn't announce the hint text.

### Expected behavior / Acceptance criteria

There are arguments for and against including hint text in the label/legend.

Whatever decision you make, it needs to be consistent across all components - either they should _all_ include it, or _all_ not include it.

### Supporting evidence (screenshots, recordings, code snippets)

va-select's hint text is below the label:

<img width="1288" height="512" alt="Image" src="https://github.com/user-attachments/assets/78f799b3-5945-44ee-bffb-3cd35e0ac8d2" />

va-text-input's hint text is within the label:

<img width="656" height="165" alt="Image" src="https://github.com/user-attachments/assets/dc90d8f3-71be-4527-9b02-e2819635d76a" />

### Type of issue

- [ ] Axe-core
- [ ] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [ ] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [x] Screenreader | assistive tech & device support
- [x] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else

**Comment by caw310** [comment]
There are some inconsistencies. Some are okay but others may need to be fixed. Jeana to confirm if we want to make the changes and what to do with va-checkbox especially.
